velox/docs/develop/types.rst

```diff
@@ -174,14 +175,31 @@ Supported range of milliseconds is [0xFFF8000000000000L, 0x7FFFFFFFFFFFF]
 store timezone ID. Supported range of timezone ID is [1, 1680].
 The definition of timezone IDs can be found in ``TimeZoneDatabase.cpp``.
 
-IPADDRESS represents an IPV6 or IPV4 formatted IPV6 address. Its physical
-type is HUGEINT. The format that the address is stored in is defined as part of `(RFC 4291#section-2.5.5.2) <https://datatracker.ietf.org/doc/html/rfc4291.html#section-2.5.5.2>`_
+IPADDRESS represents an IPv6 or IPv4 formatted IPv6 address. Its physical
+type is HUGEINT. The format that the address is stored in is defined as part of `RFC 4291#section-2.5.5.2 <https://datatracker.ietf.org/doc/html/rfc4291.html#section-2.5.5.2>`_.
 As Velox is run on Little Endian systems and the standard is network byte(Big Endian)
 order, we reverse the bytes to allow for masking and other bit operations
 used in IPADDRESS/IPPREFIX related functions. This type can be used to
 create IPPREFIX networks as well as to check IPADDRESS validity within
 IPPREFIX networks.
 
+IPPREFIX represents an IPv6 or IPv4 formatted IPv6 address along with a one byte
+prefix length. Its physical type is VARBINARY but has a fixed length of 17 bytes.
+The format that the address is stored in is defined as part of `RFC 4291#section-2.5.5.2 <https://datatracker.ietf.org/doc/html/rfc4291.html#section-2.5.5.2>`_.
+The prefix length is stored in the last byte of the VARBINARY array.
+The IP address stored is the canonical(smallest) IP address in the
+subnet range. This type can be used in IP subnet functions.
+
+Example:
+
+In this example the first 32 bits(*FFFF:FFFF*) represents the network prefix.
+As a result the IPPREFIX object stores *FFFF:FFFF::* and the length 32 for both of these IPPREFIX objects. 
+
+::
+
+   IPPREFIX 'FFFF:FFFF:FFFF:FFFF:FFFF:FFFF:FFFF:FFFF/32' -- IPPREFIX 'FFFF:FFFF:0000:0000:0000:0000:0000:0000/32'
```

Comment:
I find this confusing. If IP address in IPPrefix is the smallest canonical IP address in the subnet range, then can IPPREFIX 'FFFF:FFFF:FFFF:FFFF:FFFF:FFFF:FFFF:FFFF/32' be a valid IPPrefix ?

Reply:
Its a valid IPPREFIX, but everything in 
IPPREFIX 'FFFF:FFFF:XXXX:XXXX:XXXX:XXXX:XXXX:XXXX/32' 
All belong to the same IPPREFIX. Which is `FFFF:FFFF`. 

This is because the prefix length is only 32 bits.

Reply:
Ah I see. So conversions between IPPrefixes could be lossy in these cases. So IPPrefix -> string -> IPPrefix will not return the same IPPrefix ?

Reply:
Well I wouldn't call it lossy, as the rest of the bits to the right have no impact on the subnet. It would be something like, we want the subnet of this IP with this prefix length. 

So if you were to cast the IPPREFIX to IPADDRESS, then we are asking, what is the subnet, not what is the original IPADDRESS we used to create this subnet. 

Here is it running in Presto Java
<img width="1089" alt="Screenshot 2024-10-01 at 3 55 37 PM" src="https://github.com/user-attachments/assets/03e0cfbb-74f1-411f-8667-3a0a41537b8e">

<img width="1117" alt="Screenshot 2024-10-01 at 3 56 00 PM" src="https://github.com/user-attachments/assets/7abb0132-74f9-4f95-be10-26e6420a2290">